internal/compress/gob/gob.go

```diff
@@ -0,0 +1,39 @@
+package gob
+
+import (
+	"encoding/gob"
+	"io"
+)
+
+// Encoder represents an interface for Encoder of gob.
+type Encoder interface {
+	Encode(e interface{}) error
+}
```

Comment:
I can understand your opinion.
but if I use type alias, I can not write test code.
I want to mock `Encode` method. and I want to write a test for an error or not pattern.
https://github.com/vdaas/vald/blob/2a4322d0aa00f292fd22389136664449c5082440/internal/compress/gob.go#L104

Reply:
@rinx @hlts2 seems this conversation is resolved, @hlts2 needs interface type instead of type alias for testing, which is make sense.
So I'll merge this PR.
thank you for your help & attention !